### PR TITLE
Fix slicing out of range in units

### LINKED ISSUE #372: BUG: slicing not working as expected when out of range
Slicing with float do not always return an error when slicing outside the limit. 

For s:

>>> s = signals.Spectrum(range(1000))
>>> s.axes_manager[0].scale = 0.01
>>> s.axes_manager[0].units = "keV"

Then

>>> s[15.]

Return ``ValueError: The value is out of the axis limits`` as expected. But 

>>> s[15.:16.]

Return a spectrum but it should return a ``ValueError``

### PR DESCRIPTION
**Pull request by francisco-dlp**
Fixes #372

### CODE REVIEW

**Comment by to266** [comment]
I would expect the behaviour to be slightly different than you implemented, namely if the requested slice is a superset of the total axis, I would expect to get the full axis. 

Also, while testing this I probably found another bug:
```python
In [1]: import hyperspy.api as hs

In [2]: s = hs.datasets.example_signals.EDS_TEM_Spectrum()

In [3]: s.axes_manager[0].axis[:5] #check beginning of axis to know where to slice
Out[3]: array([ 0.169315,  0.189343,  0.209371,  0.229399,  0.249427])

In [4]: s.axes_manager[0].axis[-5:] #check end of axis to know where to slice
Out[4]: array([ 19.936951,  19.956979,  19.977007,  19.997035,  20.017063])

In [5]: s.isig[-3.:20.].axes_manager[0].axis[:5] # all as expected
Out[5]: array([ 0.169315,  0.189343,  0.209371,  0.229399,  0.249427])

In [6]: s.isig[-3.:20.].axes_manager[0].axis[-5:] 
Out[6]: array([ 19.896895,  19.916923,  19.936951,  19.956979,  19.977007])
```
I would expect the last result to be 1 pixel larger, include the 19.997035 point. Rounding error?

If we were leaving out the last point in the slice, as is accustomed in python then this is strange:
```python
In [7]: s.isig[-3.:20.015].axes_manager[0].axis[-5:] # not quite the last point, but we still get the second-last
Out[7]: array([ 19.916923,  19.936951,  19.956979,  19.977007,  19.997035])
```

**Comment by francisco-dlp** [comment]
Thanks, it should be fixed now.

Regarding the rounding issue, hyperspy simply uses python's round.

File: hyperspy/axes.py [hunk 1]
```diff
@@ -227,30 +227,34 @@ def _get_array_slices(self, slice_):
...
+                    # The start value is above the axis limit
+                    raise IndexError(
+                        "Start value above axis high bound for  axis %s."
+                        "value: %f high_bound: %f" % (repr(self), start,
+                                                      self.low_value))
                 else:
-                    stop = None
+                    # The start value is below the axis limit,
+                    # we slice from the start.
+                    start = None
```
Comment (to266) [inline hyperspy/axes.py]: this should be `stop`. Also in the error text above (start - > stop, above->below, high->low)

Comment (francisco-dlp) [inline hyperspy/axes.py]: Thanks, it should be fine now.

**Comment by vidartf** [comment]
FYI: Note that the rounding behavior of python3 is different from python2 ( 0.5 and 1.5 are rounded different ways).

**Pull request merged**